docs/content/3.middleware/2.guides/7.api-client.md

```diff
@@ -105,71 +98,91 @@ export type ContextualizedEndpoints = {
 export type MyIntegrationIntegrationContext = IntegrationContext<
   AxiosInstance, // HTTP client instance
   MiddlewareConfig,
-  ContextualizedEndpoints
+  TODO
 >;
+```
 
-/**
- * Global context of the application which includes runtime integration context.
- **/
-export interface Context {
-  // This property is named `myIntegration`, but you should use your integration name in here.
-  $myIntegration: MyIntegrationIntegrationContext;
-}
+Now, you can create the first API method - an `exampleEndpoint` function.
+
+```ts
+// api/exampleEndpoint/index.ts
+import { MyIntegrationIntegrationContext, TODO } from "../../types";
+
+export const exampleEndpoint = async (
+  context: MyIntegrationIntegrationContext,
+  params: TODO
+) => {
+  console.log("exampleEndpoint has been called");
+
+  // Example request could look like this:
+  // return await context.client.get(`example-url?id=${params.id}`);
+  return Promise.resolve({ success: true });
+};
 ```
 
-The `ContextualizedEndpoints` type is a set of API functions without the `context` argument. It's necessary for the `IntegrationContext` type. It also requires the type of client used in the integration and the type of integration configuration.
+You should also export the `exampleEndpoint` function in the `api/index.ts` file.
 
-The `Context` type is the global context of the application. It's used by api-client methods to access the integration client and the configuration.
+```ts
+// api/index.ts
+export * from "./exampleEndpoint";
+```
 
-Now you can add the interface for the API functions.
+Then, let's create the `Endpoints` type.
 
 ```ts
-// types/api/index.ts
-import { MyIntegrationContext } from '../config';
+// types/api/endpoints.ts
+import { WithoutContext } from "@vue-storefront/middleware";
+import * as apiMethods from "../../api";
 
-type TODO = any;
+export type ApiMethods = typeof apiMethods;
 
-/**
- * Definition of all API-client methods available in {@link https://docs.alokai.com/v2/advanced/context.html#context-api | context}.
- */
-export interface Endpoints {
-  /**
-   * Here you can find an example endpoint definition. Based on this example, you should define how your endpoint will look like.
-   * This description will appear in the API extractor, so try to document all endpoints added here.
-   */
-  exampleEndpoint: (context: MyIntegrationContext, params: TODO) => Promise<TODO>;
-}
+export type Endpoints = WithoutContext<ApiMethods>;
```

Comment:
just a small note: the tradeoff of using the `WithoutContext<ApiMethods` is that when the user will try to find the definition of the SDK Method within the IDE, he will be navigated to the `exampleEndpoint` method implementation, which contains context as a first argument. So this may be misleading, because the Endpoints are without context.

I would recommend another approach for implementing the integration:

1. First - specify the `Endpoints` type (API Design First Approach)
```
export type Endpoints = {
  exampleEndpoint(params: TODO): TODO_RETURN;
}
```
2. Compute the API Methods type
```
export type ApiMethods = ContextualizedEndpoints<...>;
```
3. Implement the endpoints
```
export const exampleEndpoint: ApiMethods["exampleEndpoint"] = (context, params) => {...}
```
4. Add module to the Storefront
```
customModule: middlewareModule<Endpoints>
```
Then when the user will inspect the method `sdk.customModule.exampleEndpoint`, he will navigate to the `Endpoints` type which won't contain the context as first param

Reply:
Let me rethink the proposal as it would require to refactor already refactored integrations.